node/full_client_test.go

```diff
@@ -711,6 +706,39 @@ func TestValidatorSetHandling(t *testing.T) {
 		err := nodes[i].Start()
 		require.NoError(err)
 	}
+	return rpc
+}
+
+// Tests moving from two validators to one validator and then back to two validators
+func TestValidatorSetHandling(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+
+	waitCh := make(chan interface{})
+
+	numNodes := 2
```

Comment:
[non-blocking][proposal]: Good candidate for matrix & boundaries testing scenarios in the future. 
Some ideas: 

- [ ] From 1 to N
- [ ] From N to 1 and N/2

Reply:
Thanks @Bidon15! Created separate issue for #862